### PR TITLE
Date: Add distance(to:) and advanced(by:) introduced in macOS 10.15

### PR DESCRIPTION
**Pull request by spevans**

### CODE REVIEW

**Comment by spevans** [comment]
@swift-ci test linux

File: Sources/Foundation/Date.swift [hunk 1]
```diff
@@ -190,9 +190,29 @@ public struct Date : ReferenceConvertible, Comparable, Equatable {
     public static func -=(lhs: inout Date, rhs: TimeInterval) {
         lhs = lhs - rhs
     }
-    
 }
 
+@available(OSX 10.15, iOS 13.0, tvOS 13.0, watchOS 6.0, *)
```
Comment (millenomi) [inline Sources/Foundation/Date.swift:195]: Please use `@available(swift 5.3)` for Swift.

**Comment by millenomi** [comment]
Okay to go with one change.

**Comment by spevans** [comment]
@swift-ci test linux

**Comment by spevans** [comment]
@swift-ci test linux

**Pull request merged**